### ISSUE TITLE
EIDF upload

### ISSUE THREAD

**Issue by mjaquiery** [Battery-Intelligence-Lab/galvanalyser#33]
To send data to the EIDF, Galvanalyser needs to:

- [x] monitor battery data directories
- [x] extract file and dataset metadata
- [x] allow metadata to be checked and edited by authorised users
- [ ] send raw files and metadata to 3rd parties (i.e. the EIDF's CKAN ingestion service)

This will require: 
- [ ] Authenticating with EIDF CKAN
- [ ] Implementing the EIDF CKAN ingestion API
- [ ] Providing users an action button to perform uploading
- [ ] Tracking upload status and reporting it to users (e.g. EIDF CKAN record uid)

**Comment by martinjrobins** [comment]
so a user can upload to EIDF even before the data has been imported into their galvanalyser dbase? Since you only need the raw file....

**Comment by mjaquiery** [comment]
Yes, theoretically. There was some discussion on whether files processed with Galvanalyser should also upload processed data to the EIDF. Currently, provided the metadata are processed and the dataset created, the metadata can be completed by the user and the metadata + raw files would be sent to EIDF. 

Still working on how to get the raw files to EIDF given they never reach Galvanalyser (perhaps an instruction to the Harvester to send the file with appropriate metadata?).
